### DISCUSSION TITLE
Looking for guidance resolving system lockups

### DISCUSSION THREAD

**Discussion by RobertWHurst** [openzfs/zfs#16389] [Q&A]
I've been running Ubuntu on my home server (used for some professional things too) with a ZFS raidz2 pool made up of 6 8tb Seagate BarraCuda drives - about 40tb of capacity. I'm running a number of containers for things like plex and immach as well as an instance gitlab.

My system keeps locking up about every 24 to 48 hours. Occasionally I see the following error printed to stdout `[12166.497041] INFO: task txg_sync:1475 blocked for more than 122 seconds.` followed by `[12166.497101] Tainted: P 1] 6.8.0-39-generic #39-Ubuntu` - an unhappy kernel indeed.

Initially I thought the drives were failing, so I started replacing them, but to no avail. I'm wondering if it has to do with free space. The pool has about 3 tbs left. I have another 4 drives on standby should I need to move my data off of the pool to a basic vdev temporarily to destroy and replace the pool.

I've also tried to solve the issue by switching to the latest firmware for my motherboard, and I've flashed my LSI 9211 controller with the latest IT mode firmware neither of which made a difference. I read a few GitHub issues suggesting that some linux kernels don't play nice with zfs, so I just completed a full system upgrade going from 22.04 to 24.04, thankfully the upgrade went fine, but I still have these lock ups.

I've now started a ritual of waking up at 7am to login to my pikvm and restart the machine. Any help is greatly appreciated!

**Comment by RobertWHurst** [comment]
@scineram Do you have some feedback apart from the emoji?

**Comment by allanjude** [comment]
Can you share the model number of the Barracuda drives? If they are SMR, they might actually take a long time to perform some writes/syncs since they need to re-write an entire 256 MB zone.
